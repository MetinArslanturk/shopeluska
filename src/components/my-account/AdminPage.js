import React from 'react';
import { connect } from 'react-redux';
import ActionButton from '../ActionButton';
import AddUpdateProductModal from '../AddUpdateProductModal';
import { Table } from 'antd';

export class AdminPage extends React.Component {

    state = {
        addProductOpen: false,
        isUpdate: false,
        productData: {}
    };


    productColumns = [
        {
          title: 'Name',
          dataIndex: 'name'
        },
        {
          title: 'Description',
          dataIndex: 'description'
        },
        {
            title: 'Price',
            dataIndex: 'price'
        },
        {
          title: 'Stock',
          dataIndex: 'stock'
        },
        {
          title: 'Actions',
          dataIndex: 'action',
          render: (text, record) => (
              <span>
                  <ActionButton
                      text="Edit"
                      onClickActionFunction={this.openAddUpdateProductModal}
                      itemToArguments={record}
                  />

                  <ActionButton
                      text="Delete"
                      onClickActionFunction={this.openAddUpdateProductModal}
                      itemToArguments={record}
                  />
              </span>
          ),
        },
      ];


    handleClose = () => {
        this.setState(() => ({ addProductOpen: false }));
    }

    handleSave = (savedProduct) => {
        this.setState(() => ({ addProductOpen: false }));
    }

    openAddUpdateProductModal = (record) => {
            this.setState(() => ({ addProductOpen: true, isUpdate: !!record, productData: record }));
    }

    render() {
        return (
            <>
                Manage Products <br />

                <div className="align-to-right">
                    <ActionButton
                        text="Add New Product"
                        type="primary"
                        onClickActionFunction={this.openAddUpdateProductModal}
                        itemToArguments={undefined}
                    />
                </div>

                <AddUpdateProductModal
                    isOpen={this.state.addProductOpen}
                    isUpdate={this.state.isUpdate}
                    productData={this.state.productData}
                    handleSave={this.handleSave}
                    handleClose={this.handleClose}>
                </AddUpdateProductModal>

                <Table columns={this.productColumns} dataSource={this.props.products} />
            </>
        )
    }
}

const mapStateToProps = (state) => ({
    products: state.products.products
})

export default connect(mapStateToProps)(AdminPage);