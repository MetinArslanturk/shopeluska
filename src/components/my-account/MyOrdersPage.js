import React from 'react';
import { connect } from 'react-redux'
import { Table } from 'antd';

export class MyOrdersPage extends React.Component {
    columns = [
        {
          title: 'Name',
          dataIndex: 'name',
          render: text => <>{text}</>,
        },
        {
          title: 'Age',
          dataIndex: 'age'
        },
        {
          title: 'Address',
          dataIndex: 'address'
        },
        {
          title: 'Action',
          dataIndex: 'action',
          render: (text, record) => (
            <span>
              <button style={{ marginRight: 16 }} onClick={this.clickd}>Invite {record.name}</button>
            </span>
          ),
        },
      ];
      data = [
        {
          key: '1',
          name: 'John Brown',
          age: 32,
          address: 'New York No. 1 Lake Park',
          tags: ['nice', 'developer'],
        },
        {
          key: '2',
          name: 'Jim Green',
          age: 42,
          address: 'London No. 1 Lake Park',
          tags: ['loser'],
        },
        {
          key: '3',
          name: 'Joe Black',
          age: 32,
          address: 'Sidney No. 1 Lake Park',
          tags: ['cool', 'teacher'],
        },
      ];

    clickd = () => {
        console.log('here');
    }
    render() {
        return (
            <>
                My Orders page.
                <Table columns={this.columns} dataSource={this.data} />
            </>
        )
    }
}

export default connect(undefined)(MyOrdersPage);