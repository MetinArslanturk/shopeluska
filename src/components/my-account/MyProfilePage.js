import React from 'react';
import { connect } from 'react-redux';

export class MyProfilePage extends React.Component {
    render() {
        return (
            <>
                My Profile page.
            </>
        )
    }
}

export default connect(undefined)(MyProfilePage);