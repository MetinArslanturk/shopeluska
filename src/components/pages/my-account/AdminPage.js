/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Table } from 'antd';
import { startDeleteProduct } from '../../../actions/products';
import { productColumns } from '../../../helpers/productColumns';
import ActionButton from '../../common-components/ActionButton';
import { deleteConfirm } from '../../common-components/ConfirmPopup';
import AddUpdateProductModal from '../../common-components/AddUpdateProductModal';

export class AdminPage extends React.Component {
    state = {
        addProductOpen: false,
        isUpdate: false,
        productData: {},
    };

    columns = [
        ...productColumns,
        {
            title: 'Actions',
            dataIndex: 'action',
            render: (text, record) => (
                <span>
                    <ActionButton
                        text="Edit"
                        onClickActionFunction={this.openAddUpdateProductModal}
                        itemToArguments={record}
                    />

                    <ActionButton
                        text="Delete"
                        id="delete-button"
                        onClickActionFunction={this.handleDelete}
                        itemToArguments={record}
                    />
                </span>
            ),
        },
    ];

    handleDeleteConfirm = (id) => {
        this.props.startDeleteProduct(id);
    };

    handleDelete = (record) => {
        deleteConfirm(this.handleDeleteConfirm, record._id);
    };

    handleClose = () => {
        this.setState(() => ({ addProductOpen: false }));
    };

    handleSave = (/* savedProduct */) => {
        this.setState(() => ({ addProductOpen: false }));
    };

    openAddUpdateProductModal = (record) => {
        this.setState(() => ({
            addProductOpen: true,
            isUpdate: !!record,
            productData: record,
        }));
    };

    render() {
        return (
            <>
                <h2>Manage Products</h2> <br />
                <div className="align-to-right">
                    <ActionButton
                        text="Add New Product"
                        type="primary"
                        onClickActionFunction={this.openAddUpdateProductModal}
                        itemToArguments={undefined}
                    />
                </div>
                <AddUpdateProductModal
                    isOpen={this.state.addProductOpen}
                    isUpdate={this.state.isUpdate}
                    productData={this.state.productData}
                    handleSave={this.handleSave}
                    handleClose={this.handleClose}
                />
                <Table
                    columns={this.columns}
                    dataSource={this.props.products}
                />
            </>
        );
    }
}

const mapStateToProps = (state) => ({
    products: state.products.products,
});

const mapDispatchToProps = (dispatch) => ({
    startDeleteProduct: (id) => dispatch(startDeleteProduct(id)),
});

AdminPage.propTypes = {
    products: PropTypes.array,
    startDeleteProduct: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(AdminPage);
