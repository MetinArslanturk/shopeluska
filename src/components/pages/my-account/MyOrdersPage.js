import React, { useEffect, useMemo } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Table } from 'antd';
import { getOrdersAction } from '../../../actions/shopping';

export const MyOrdersPage = ({ myOrders, getOrders }) => {
    useEffect(() => {
        getOrders();
    }, [getOrders]);

    const orders = useMemo(() => {
        return myOrders.map((order, i) => {
            let totalPrice = 0;
            const productNames = [];
            order.orders.forEach((element) => {
                totalPrice += element.orderPrice;
                productNames.push(element.product.name);
            });
            return {
                createdAt: order.orderUser.createdAt,
                productNames: productNames.join(', '),
                totalPrice,
                key: order.orderUser.createdAt + i,
            };
        });
    }, [myOrders]);

    const columns = [
        {
            title: 'Products',
            dataIndex: 'productNames',
            key: 'productNames',
        },
        {
            title: 'Date',
            dataIndex: 'createdAt',
            key: 'createdAt',
            render: (text) => <span>{new Date(text).toLocaleString()}</span>,
        },
        {
            title: 'Total Price',
            dataIndex: 'totalPrice',
            key: 'totalPrice',
        },
    ];

    return (
        <>
            <Table columns={columns} dataSource={orders} />
        </>
    );
};

MyOrdersPage.propTypes = {
    myOrders: PropTypes.array,
    getOrders: PropTypes.func,
};

const mapStateToProps = (state) => ({
    myOrders: state.shopping.orders,
});

const mapDispatchToProps = (dispatch) => ({
    getOrders: () => dispatch(getOrdersAction()),
});

export default connect(mapStateToProps, mapDispatchToProps)(MyOrdersPage);
