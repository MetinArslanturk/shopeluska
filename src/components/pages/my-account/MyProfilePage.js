import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Form, Input, Button } from 'antd';
import { getUserName, getEmail } from '../../../selectors/auth';
import { startUpdateMyProfile } from '../../../actions/auth';

export class MyProfilePage extends React.Component {
    layout = {
        labelCol: {
            span: 8,
        },
        wrapperCol: {
            span: 16,
        },
    };

    tailLayout = {
        wrapperCol: {
            offset: 8,
            span: 16,
        },
    };

    onFinish = (values) => {
        // eslint-disable-next-line no-shadow
        const { startUpdateMyProfile } = this.props;
        startUpdateMyProfile(values);
    };

    render() {
        const { username, email } = this.props;
        return (
            <>
                <div className="login-wrapper">
                    <Form
                        {...this.layout}
                        name="basic"
                        initialValues={{
                            username,
                            email,
                        }}
                        className="user-info-wrapper"
                        onFinish={this.onFinish}
                    >
                        <Form.Item label="Username" name="username">
                            <Input />
                        </Form.Item>

                        <Form.Item label="Password" name="password">
                            <Input.Password />
                        </Form.Item>

                        <Form.Item label="Email" name="email">
                            <Input />
                        </Form.Item>

                        <Form.Item {...this.tailLayout}>
                            <Button type="primary" htmlType="submit">
                                Submit
                            </Button>
                        </Form.Item>
                    </Form>
                </div>
            </>
        );
    }
}

MyProfilePage.propTypes = {
    startUpdateMyProfile: PropTypes.func,
    username: PropTypes.string,
    email: PropTypes.string,
};

const mapStateToProps = (state) => ({
    username: getUserName(state.auth.user),
    email: getEmail(state.auth.user),
});

const mapDispatchToProps = (dispatch) => ({
    startUpdateMyProfile: (values) => dispatch(startUpdateMyProfile(values)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MyProfilePage);
