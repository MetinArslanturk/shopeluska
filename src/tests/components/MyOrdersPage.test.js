import React from 'react';
import { mount, shallow } from 'enzyme';
import { MyOrdersPage } from '../../components/pages/my-account/MyOrdersPage';

const myOrders = [
    {
        orderUser: { _id: 'testid', user: 'testuser', createdAt: 10000 },
        orders: [
            {
                _id: 'order',
                orderUser: 'testid',
                product: { _id: 'prodid', name: 'prod name' },
                key: '123',
            },
        ],
    },
];

test('should render', () => {
    const wrapper = shallow(<MyOrdersPage myOrders={myOrders} />);
    expect(wrapper).toMatchSnapshot();
});

test('should call getOrders', (done) => {
    const getOrders = jest.fn().mockImplementation(() => {
        expect(getOrders).toHaveBeenCalled();
        done();
    });
    mount(<MyOrdersPage getOrders={getOrders} myOrders={myOrders} />);
});
