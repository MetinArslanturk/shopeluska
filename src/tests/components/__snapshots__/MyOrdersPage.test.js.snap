// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<Fragment>
  <Table
    columns={
      Array [
        Object {
          "dataIndex": "productNames",
          "key": "productNames",
          "title": "Products",
        },
        Object {
          "dataIndex": "createdAt",
          "key": "createdAt",
          "render": [Function],
          "title": "Date",
        },
        Object {
          "dataIndex": "totalPrice",
          "key": "totalPrice",
          "title": "Total Price",
        },
      ]
    }
    dataSource={
      Array [
        Object {
          "createdAt": 10000,
          "key": 10000,
          "productNames": "prod name",
          "totalPrice": NaN,
        },
      ]
    }
    rowKey="key"
  />
</Fragment>
`;
